code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -98,7 +98,8 @@
 		if(target.reagents.total_volume >= target.reagents.maximum_volume)
 			to_chat(user, "<span class='warning'>[target] is full.</span>")
 			return
-
+		if(isextinguisher(target) && !src.reagents.only_reagent("aqueous_foam"))
```

Comment:
может лучше всё же позволить заправляться любым танком? так будет минимальная хоть грифозность.

ну и проверять что в танке всего один реагент. если реагентов не сколько - условно говоря нет нужного давления в танке чтобы заправить огнетушитель!

Reply:
ничего не понял, но понял что Людук грифер.

Reply:
> предложение:
> 
> Огнетушитель заправляется только из water tank, fuel tank, afff tank. Если в них есть больше чем одного реагента (кто-то что-то подмешал), огнетушитель не заправляется. Писать сообщение "You can't refill extinguisher, because [tank] doesn't have enough pressure."
> 
> Таким образом:
> 
> * сохраняем фичу про наполнение
> * даём возможность наполнять чем-то грифозным типо воды, или делать мейкшифт огнемёт
> * имеем полный контроль над тем что именно можно заполнить в огнетушитель (к примеру можно будет запретить наполняться из бака с топливом если будет СЛИШКОМ грифозно)
> 
> на сколько я знаю смазка и перцовка главные кандидаты на грифоз огнетушителем (не считая бабаха) и потому таким решением будут отсечены.

Все это предложение ломается о то, что эти самые баки можно наполнить чем угодно, а опустошить их довольно легко. Т.е. тут на гите мы уже делаем по сути вайтлист реагентов, какими разрешаем пользоваться, а какими нет.

*(И да, зачем огнемет через разливание топлива из огнетушителя, привет 451, если есть нормальный огнемет, который можно или напечатать или скрафтить, который только из-за текущего своего состояния мало используется.)* 

И кроме перца и тд. есть ещё политринник, или серная, металлическая пена. На станции все ещё есть хим спреи которые менее грифозные. 

*(Хотя даже ту же воду используют не для тушения пожаров, в чем она к слову, достаточно эффективна, а для создания мокрого пола.)*

В таком случае фиксится только бумпалка, да и то, я думаю можно будет придумать как заабузить чтобы она снова работала.

Можем тут или на форуме решить, что можно будет заливать в огнетушитель, а что нет, изменю под это чем-то вроде того, что уже есть, просто проверяя не на 1 единственный реагент, а на реагент из списка и если есть хотя бы один реагент, который не входит в список, запрещать наливать в огнетушитель. *(По хорошему, вообще, нужно бы сделать чтобы оно как в машинерии вирусологии, забирало только нужное, но это по желанию, если его проявят.)*

Reply:
Есть возможность определять какой реагент трансферить, а какой нет. Никаких проблем от этого метода не указано. Заливать по хорошему надо иметь возможность всё. Другое дело, что не все реагенты реализованы в виде каких-то уникальных жидкостей, кроме тех, на которые предлагается вайтлист. И ограничить всё только пеной убивает дух исследований.

Например, мало кто задумывается, но пену и топливо можно смешать вместе в огнетушителе и они будут работать по-своему, не так как отдельно. В баках их так смешивать невозможно. Ты убиваешь именно такие фичи, если не понял.

Reply:
Это все ещё сводит все обсуждение к тому, что "мы хотим вайтлист для грифф реагентов, но не слишком грифозных, по нашему мнению".

И да. Для исследований есть профа ученого. *(Сандбокс)* Сиди после бомб и с хим спреем развлекайся сколько влезет, к чему тут огнетушитель.

Reply:
Нет, никто не хочет вайтлист. У тебя ложное восприятие. Проблем даже с бомбами, которые надо решать, нет ни у кого.

Reply:
ок